pkg/chunkenc/unordered.go

```diff
@@ -135,7 +135,7 @@ func (hb *unorderedHeadBlock) Append(ts int64, line string, structuredMetadata l
 		for _, et := range displaced[0].(*nsEntries).entries {
 			if et.line == line {
 				e.entries = displaced[0].(*nsEntries).entries
-				return nil
+				return ErrDuplicateEntry
```

Comment:
I don't think you can do this? returning an Error here I believe will be propagated as a failure of the push of the line, which is a change in semantics as to how Loki processes exact dupes (we currently do not fail on this, rather silently swallow the duplicates)

Reply:
Instead of incrementing the metric in 2 other places and returning an error, can we increment the metric here? This is really the place were exact dupes are actually handled (at least when running Loki with out of order ingest)